saleor/dashboard/product/views.py

```diff
@@ -527,3 +528,36 @@ def stock_location_delete(request, location_pk):
     return TemplateResponse(
         request, 'dashboard/product/stock_location/modal_confirm_delete.html',
         ctx)
+
+
+@require_POST
+@staff_member_required
+def ajax_reorder_product_images(request, product_pk):
+    product = get_object_or_404(Product, pk=product_pk)
+    data = json.loads(request.body)
```

Comment:
I think we could use custom form that would accept one field with comma separated integers. Then you could move this whole logic there and keep the view simple.